url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/treemaker/XTreeMaker.java

```diff
@@ -0,0 +1,1486 @@
+			excType.add(makeExprPrefixSuffix(false));
+			while(tokenOk(XTokenKind.KEYWORD, XKeyword.OR)){
+				tokenizer.next();
+				excType.add(makeExprPrefixSuffix(false));
+			}
+			XTreeIdent varName = null;
+			if(!tokenOk(XTokenKind.KEYWORD, XKeyword.RBRAKET)){
+				varName = makeIdent();
+			}
+			expect(XKeyword.RBRAKET);
+			XTreeScope b = makeScope();
+			catches.add(new XTreeCatch(endPosition(), excType, varName, b));
+			startPosition();
+		}
+		endPosition();
+		XTreeScope _finally;
+		if(tokenOk(XTokenKind.KEYWORD, XKeyword.FINALLY)){
+			tokenizer.next();
+			_finally = makeScope();
+		}else{
+			_finally = null;
+		}
+		return new XTreeTry(endPosition(), resources, body, catches, _finally);
+	}
+	
+	private XTreeMakeList makeMakeListPosStarted(){
+		expect(XKeyword.LINDEX);
+		List<XTreeExpr> exprs = new ArrayList<XTreeExpr>();
+		if(!tokenOk(XTokenKind.KEYWORD, XKeyword.RINDEX) && !tokenEof()){
+			exprs.add(makeExpr(true));
+			while(tokenOk(XTokenKind.KEYWORD, XKeyword.COMMA)){
+				tokenizer.next();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.RINDEX)){
+					break;
+				}
+				exprs.add(makeExpr(true));
+			}
+		}
+		expect(XKeyword.RINDEX);
+		return new XTreeMakeList(endPosition(), exprs);
+	}
+	
+	private XTreeMakeMap makeMakeMapPosStarted(){
+		expect(XKeyword.LSCOPE);
+		List<XTreeMapEntry> entries = new ArrayList<XTreeMapEntry>();
+		if(!tokenOk(XTokenKind.KEYWORD, XKeyword.RSCOPE) && !tokenEof()){
+			entries.add(makeMapEntry());
+			while(tokenOk(XTokenKind.KEYWORD, XKeyword.COMMA)){
+				tokenizer.next();
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.RSCOPE)){
+					break;
+				}
+				entries.add(makeMapEntry());
+			}
+		}
+		expect(XKeyword.RSCOPE);
+		return new XTreeMakeMap(endPosition(), entries);
+	}
+	
+	private XTreeMapEntry makeMapEntry(){
+		startPosition();
+		XTreeExpr key = makeExpr();
+		expect(XKeyword.COLON);
+		XTreeExpr expr = makeExpr(true);
+		return new XTreeMapEntry(endPosition(), key, expr);
+	}
+	
+	private XTreeCall makeCallPosStarted(XTreeExpr method){
+		expect(XKeyword.LBRAKET);
+		List<XTreeExpr> params = new ArrayList<XTreeExpr>();
+		List<XTreeIdent> kws = new ArrayList<XTreeIdent>();
+		if(!tokenOk(XTokenKind.KEYWORD, XKeyword.RBRAKET) && !tokenEof()){
+			XTreeExpr e = makeExpr(true);
+			if(tokenOk(XTokenKind.KEYWORD, XKeyword.COLON) && e instanceof XTreeIdent){
+				tokenizer.next();
+				kws.add((XTreeIdent)e);
+				e = makeExpr(true);
+			}else{
+				kws.add(null);
+			}
+			params.add(e);
+			while(tokenOk(XTokenKind.KEYWORD, XKeyword.COMMA)){
+				tokenizer.next();
+				e = makeExpr(true);
+				if(tokenOk(XTokenKind.KEYWORD, XKeyword.COLON) && e instanceof XTreeIdent){
+					tokenizer.next();
+					kws.add((XTreeIdent)e);
+					e = makeExpr(true);
+				}else{
+					kws.add(null);
+				}
+				params.add(e);
+			}
+		}
+		expect(XKeyword.RBRAKET);
+		return new XTreeCall(endPosition(), method, params, kws);
+	}
+	
+	public boolean isOperator(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1340-1349 into a separate method.